src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -308,24 +285,78 @@ grpc_channel_credentials* grpc_google_default_credentials_create() {
     g_metadata_server_available = g_is_on_gce;
```

Comment:
sorry also while we're here, we're writing to `g_is_on_gce` with `g_state_mu`, but reading from it without it, can we fix that here too?

Reply:
The only unprotected access left is the focus of an issue under discussion with @yihuazhang. The resolution might be to remove it entirely, so I'll wait on this change until we hear back from him.

Reply:
The last reference has been removed, so there are no unprotected accesses left.